repose-aggregator/commons/commons-utilities/src/main/scala/org/openrepose/commons/utils/servlet/http/HttpServletResponseWrapper.scala

```diff
@@ -333,13 +390,20 @@ class HttpServletResponseWrapper(originalResponse: HttpServletResponse, headerMo
       throw new IllegalStateException("Cannot call resetBuffer after the response has been committed")
     } else {
       super.resetBuffer()
+      // try to remove the content-type and content-length headers (best-effort)
+      Try {
+        removeHeader(CommonHttpHeader.CONTENT_TYPE.toString)
+        removeHeader(CommonHttpHeader.CONTENT_LENGTH.toString)
+      }
+      Try {
+        setHeader(CommonHttpHeader.CONTENT_TYPE.toString, null)
+        setHeader(CommonHttpHeader.CONTENT_LENGTH.toString, null)
+      }
+      responseBodyType = ResponseBodyType.Available
       bodyOutputStream.resetBuffer()
     }
   }
 
-  /**
-    * @throws IllegalStateException when neither headerMode nor bodyMode is ResponseMode.MUTABLE
```

Comment:
Fair enough. I'm trying to remember why I wanted to remove it, but I super can't. I want to say it would have been helpful if the Scripting filter conditionally chose a `ResponseMode` since it would prevent us from having to conditionally call this method.

Will re-add.